File: journaling-app-backend/populate_database.js
const sqlite3 = require('sqlite3').verbose();
const bcrypt = require('bcrypt');

// Initialize the database
const db = new sqlite3.Database('database.db');

db.serialize(() => {
    // Sample users
    const users = [
        { username: 'user1', email: 'user1@example.com', password: 'password1' },
        { username: 'user2', email: 'user2@example.com', password: 'password2' },
        { username: 'user3', email: 'user3@example.com', password: 'password3' },
    ];

    // Hash passwords and insert users
    users.forEach(user => {
        const hashedPassword = bcrypt.hashSync(user.password, 10);
        db.run(`
            INSERT INTO users (username, email, password)
            VALUES (?, ?, ?)
        `, [user.username, user.email, hashedPassword]);
    });

    // Sample journal entries
    const entries = [
        { user_id: 1, title: 'First Entry', content: 'This is the first entry', category: 'Personal', date: '2024-07-01' },
        { user_id: 1, title: 'Second Entry', content: 'This is the second entry', category: 'Work', date: '2024-07-02' },
        { user_id: 2, title: 'Third Entry', content: 'This is the third entry', category: 'Travel', date: '2024-07-03' },
    ];

    // Insert journal entries
    entries.forEach(entry => {
        db.run(`
            INSERT INTO journal_entries (user_id, title, content, category, date)
            VALUES (?, ?, ?, ?, ?)
        `, [entry.user_id, entry.title, entry.content, entry.category, entry.date]);
    });
});

db.close();
console.log('Database populated with sample data.');

File: backend/populate_database.py
import sqlite3
import bcrypt

def populate_db():
    conn = sqlite3.connect('database.db')
    c = conn.cursor()

    # Insert sample users
    users = [
        ('john', 'john@example.com', bcrypt.hashpw('john'.encode('utf-8'), bcrypt.gensalt())),
        ('mwirigi', 'mwirigi@example.com', bcrypt.hashpw('mwirigi'.encode('utf-8'), bcrypt.gensalt())),
        ('mahugu', 'mahugu@example.com', bcrypt.hashpw('mahugu'.encode('utf-8'), bcrypt.gensalt()))
    ]
    c.executemany('INSERT INTO users (username, email, password) VALUES (?, ?, ?)', users)

    # Get user IDs
    c.execute('SELECT id FROM users')
    user_ids = [row[0] for row in c.fetchall()]

    # Insert sample journal entries
    journal_entries = [
        (user_ids[0], 'Title 1', 'Content 1', 'Personal', '2023-01-01'),
        (user_ids[0], 'Title 2', 'Content 2', 'Work', '2023-01-02'),
        (user_ids[1], 'Title 3', 'Content 3', 'Travel', '2023-01-03'),
        (user_ids[1], 'Title 4', 'Content 4', 'Personal', '2023-01-04'),
        (user_ids[2], 'Title 5', 'Content 5', 'Work', '2023-01-05'),
        (user_ids[2], 'Title 6', 'Content 6', 'Travel', '2023-01-06')
    ]
    c.executemany('INSERT INTO journal_entries (user_id, title, content, category, date) VALUES (?, ?, ?, ?, ?)', journal_entries)

    conn.commit()
    conn.close()

if __name__ == "__main__":
    populate_db()
    print("Journal Database populated with sample records.")
